upload/system/Login/Steam.php

```diff
@@ -238,60 +248,52 @@ public function canChange($type, \IPS\Member $member)
 		return false;
 	}
 
+	/**
+	 * Gets a url param and sanitizes it for openid
+	 *
+	 * @param    string    $param  url param
+	 *
+	 * @return    string
+	 */
+	private function get($param)
+	{
+		return str_replace(' ', '+', trim(urldecode(\IPS\Request::i()->{$param})));
+	}
+
 	/**
 	 * This will validate the incoming Steam OpenID request
 	 *
 	 * @package Steam Community API
 	 * @copyright (c) 2010 ichimonai.com
 	 * @license http://opensource.org/licenses/mit-license.php The MIT License
 	 *
-	 * @return int|bool
+	 * @return bool
 	 */
 	protected function validate()
 	{
 		$params = array(
-			'openid.signed' => \IPS\Request::i()->openid_signed,
-			'openid.sig' => str_replace(' ', '+', \IPS\Request::i()->openid_sig),
-			'openid.ns' => 'http://specs.openid.net/auth/2.0',
+			'openid.assoc_handle' => $this->get('openid_assoc_handle'),
+			'openid.signed' => $this->get('openid_signed'),
+			'openid.sig' => $this->get('openid_sig'),
+			'openid.mode' => 'check_authentication'
```

Comment:
I'm assuming that this isn't functional change, just code cleanup?

Reply:
Just cleanup.

Reply:
👍